pkg/controller/datadogagent/clusteragent.go

```diff
@@ -484,7 +484,7 @@ func getEnvVarsForClusterAgent(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Env
 	}
 
 	// Cluster Checks config
-	if *spec.ClusterAgent.Config.ClusterChecksEnabled {
+	if spec.ClusterAgent.Config.ClusterChecksEnabled != nil && *spec.ClusterAgent.Config.ClusterChecksEnabled {
```

Comment:
can we use `datadoghqv1alpha1.BoolValue()` here

Reply:
Yes we can !